[Doc] Document the top-k default scheduling strategy (#32331)
ray-project/ray: doc/source/ray-core/scheduling/index.rst

"DEFAULT"
~~~~~~~~~

``"DEFAULT"`` is the default strategy used by Ray.
With the current implementation, Ray calculates a score for each node based on the node's logical resource utilization.
If the utilization is below a threshold, the score is 0. Otherwise, the score is the resource utilization itself (score 1 means the node is fully utilized).

> Ray randomly picks a node from the top k nodes with the lowest scores as the best node for scheduling.

Reviewer: do we want to also document the two options control the top_k?

"DEFAULT"
~~~~~~~~~

``"DEFAULT"`` is the default strategy used by Ray. With the current implementation, Ray will try to pack tasks or actors on nodes

> until the resource utilization is beyond a certain threshold and spread them afterwards.

Reviewer 2: Can we keep some high-level description of the policy? The new version of the text is very low-level and it's a bit hard to understand the high-level behavior, which is what most users will care about.

"DEFAULT"
~~~~~~~~~

``"DEFAULT"`` is the default strategy used by Ray.
Ray schedules tasks or actors onto a group of the top k nodes.

> Specially, the nodes are sorted to first favor those that already have tasks or actors scheduled (for locality),

Reviewer 2: ```suggestion
Specifically, the nodes are sorted to first favor those that already have tasks or actors scheduled (for locality),
```

then to favor those that have low resource utilization (for load balancing).
Within the top k group, nodes are chosen randomly to further improve load-balancing and mitigate delays from cold-start in large clusters.

Implementation-wise, Ray calculates a score for each node in a cluster based on the utilization of its logical resources.

> If the utilization is below a threshold (controlled by env var ``RAY_scheduler_spread_threshold``, default is 0.5), the score is 0,

Reviewer 2: ```suggestion
If the utilization is below a threshold (controlled by the OS environment variable ``RAY_scheduler_spread_threshold``, default is 0.5), the score is 0,
```

otherwise it is the resource utilization itself (score 1 means the node is fully utilized).
Ray selects the best node for scheduling by randomly picking from the top k nodes with the lowest scores.

> The value of ``k`` equals to the max of (# nodes in the cluster * ``RAY_scheduler_top_k_fraction`` env var) and ``RAY_scheduler_top_k_absolute`` env var.

Reviewer 2: ```suggestion
The value of ``k`` is the max of (number of nodes in the cluster * ``RAY_scheduler_top_k_fraction`` environment variable) and ``RAY_scheduler_top_k_absolute`` environment variable.
```
